src/sidebar/components/test/thread-list-test.js

```diff
@@ -1,265 +1,284 @@
-        offscreenUpperHeight: 20,
-        visibleThreads: thread.children,
-      });
-    };
-    this.detach = sinon.stub();
-    this.yOffsetOf = function () {
-      return 42;
+
+    fakeScrollContainer = document.createElement('div');
+    fakeScrollContainer.className = 'js-thread-list-scroll-root';
+    fakeScrollContainer.style.height = '2000px';
+    document.body.appendChild(fakeScrollContainer);
+
+    fakeStore = {
+      clearSelection: sinon.stub(),
     };
-    this.calculateVisibleThreads = () => {
-      return {
-        offscreenLowerHeight: 10,
-        offscreenUpperHeight: 20,
-        visibleThreads: thread.children,
-      };
+
+    fakeTopThread = {
+      id: 't0',
+      annotation: { $tag: 'myTag0' },
+      children: [
+        { id: 't1', children: [], annotation: { $tag: 't1' } },
+        { id: 't2', children: [], annotation: { $tag: 't2' } },
+        { id: 't3', children: [], annotation: { $tag: 't3' } },
+        { id: 't4', children: [], annotation: { $tag: 't4' } },
+      ],
     };
-  }
-}
-
-describe('threadList', function () {
-  let threadListContainers;
-
-  function createThreadList(inputs) {
-    const defaultInputs = {
-      thread: threadFixtures.thread,
-      onForceVisible: sinon.stub(),
-      onFocus: sinon.stub(),
-      onSelect: sinon.stub(),
-      onSetCollapsed: sinon.stub(),
+
+    fakeVisibleThreadsUtil = {
+      calculateVisibleThreads: sinon.stub().returns({
+        visibleThreads: fakeTopThread.children,
+        offscreenUpperHeight: 400,
+        offscreenLowerHeight: 600,
+      }),
+      THREAD_DIMENSION_DEFAULTS: {
+        defaultHeight: 200,
+      },
     };
 
-    // Create a scrollable container for the `<thread-list>` so that scrolling
-    // can be tested.
-    const parentEl = document.createElement('div');
-    parentEl.classList.add('js-thread-list-scroll-root');
-    parentEl.style.overflow = 'scroll';
-    parentEl.style.height = '100px';
-
-    // Add an element inside the scrollable container which is much taller than
-    // the container, so that it actually becomes scrollable.
-    const tallDiv = document.createElement('div');
-    tallDiv.style.height = '1000px';
-    parentEl.appendChild(tallDiv);
-
-    document.body.appendChild(parentEl);
-
-    // Create the `<thread-list>` instance
-    const element = util.createDirective(
-      document,
-      'threadList',
-      Object.assign({}, defaultInputs, inputs),
-      {}, // initialScope
-      '', // initialHtml
-      { parentElement: parentEl }
-    );
+    $imports.$mock(mockImportedComponents());
+    $imports.$mock({
+      'lodash.debounce': fakeDebounce,
+      '../store/use-store': callback => callback(fakeStore),
+      '../util/annotation-metadata': fakeMetadata,
+      '../util/dom': fakeDomUtil,
+      '../util/visible-threads': fakeVisibleThreadsUtil,
+    });
+  });
 
-    element.parentEl = parentEl;
+  afterEach(() => {
+    $imports.$restore();
+    fakeScrollContainer.remove();
```

Comment:
> Since this component attaches event listeners to the window, I think it will be important to ensure that it is properly unmounted by calling wrapper.unmount().

Do you mean `unmount`-ing every `wrapper` instance (i.e. within tests)? There isn't a single `wrapper` reference at present, so I'm trying to understand how I'd go about this.

Reply:
Make `createThreadList` push a reference to the `wrapper` instance onto an array (which I'll call `wrappers` here) and then call `wrappers.forEach(w => w.unmount())` in an `afterEach`.